lib/scope.js

```diff
@@ -110,15 +114,11 @@ AST_Toplevel.DEFMETHOD("figure_out_scope", function(options){
     var in_destructuring = null;
     var in_export;
     var tw = new TreeWalker(function(node, descend){
-        var create_a_block_scope =
-            (options.screw_ie8 && node instanceof AST_Catch) ||
```

Comment:
That is master code, maybe there are some issues there too. The lack of tests made the situation on harmony a bit worse though. But most important for now is to collect information about the "damage".